caikit/runtime/http_server/http_server.py

```diff
@@ -120,7 +121,13 @@ def __init__(self, tls_config_override: Optional[aconfig.Config] = None):
         super().__init__(get_config().runtime.http.port, tls_config_override)
 
         self.app = FastAPI()
-
+        self.app.add_middleware(
+            CORSMiddleware,
+            allow_origins=["*"],
+            allow_credentials=True,
+            allow_methods=["*"],
+            allow_headers=["*"],
+        )
```

Comment:
Thanks @itaijj!

I think the two things that need to happen here are:
- This whole block should be behind an `if` statement that checks a config item to see if the user wants cors middleware at all
- The four values in here should be pulled from configuration instead of using `*` and `True`

Take a look at the `caikit/config/config.yml` file, that's where the default values for the configuration returned by `caikit.get_config()` are stored. I think we'd probably want something that looks like this:

```
runtime:
  http:
    cors_middleware:
      enabled: False
      allow_origins: "*"
      allow_credentials: True
      allow_methods: "*"
      allow_headers: "*"
```

and then use the configuration values here in the code.


It would also be great to get two test cases added- one where a cross-origin request fails and one where we turn this feature on and it passes. That said I don't know enough about CORS to know if that's even possible to do in a unit test, so I wouldn't block this PR on not having tests if that's the case.